pelican/plugins/jinja_filters/jinja_filters.py

```diff
@@ -111,7 +133,23 @@ def merge_date_url(value, url):
         string: combined URL
 
     """
-    return url.format(date=value)
+    try:
+        return url.format(date=value)
+    except ValueError:
+        # will throw a "ValueError" if the value is a datetime.datetime and the
+        # url contains a "-" (e.g. "%-d") (used in Pelican to strip the leading
+        # zero)
```

Comment:
Perhaps wrap this comment to 88 characters instead of 79?

Reply:
Done!